cs-multi-shell/src/app/vault.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@angular/core';
+import { Capacitor } from '@capacitor/core';
+import {
+    BiometricSecurityStrength,
+    BrowserVault, Device, DeviceSecurityType,
+    IdentityVaultConfig, Vault, VaultError, VaultErrorCodes, VaultType
+} from '@ionic-enterprise/identity-vault';
+import { RouteService } from './route.service';
+import { AuthResult } from '@ionic-enterprise/auth';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class VaultService {
+    config: IdentityVaultConfig = {
+        key: 'io.ionic.multauth',
+        type: VaultType.DeviceSecurity,
+        deviceSecurityType: DeviceSecurityType.Biometrics,
+        lockAfterBackgrounded: 2000,
+        shouldClearVaultAfterTooManyFailedAttempts: false,
+        customPasscodeInvalidUnlockAttempts: 10,
+        unlockVaultOnLoad: false,
+    };
+
+    vault!: Vault | BrowserVault;
+
+    constructor(private routeService: RouteService) {
+    }
+
+    /**
+     * Init is called by our APP_INITIALIZER at the startup of the application
+     */
+    public async init() {
+        if (Capacitor.getPlatform() === 'web') {
+            this.vault = new BrowserVault();
+        } else {
+            // If the device does not have biometrics the we'll use a Secure Storage Vault
+            if (!await this.hasBiometrics()) {
```

Comment:
But if we dont have biometrics we'll store on the keychain/keystore using SecureStorage type.